tools/eval_plots_stackedbar.py

```diff
@@ -0,0 +1,324 @@
+    else:
+        metrics = csv_df
+
+    # Group by benchmark source
+    benchmark_by_source = metrics.groupby(['benchmark_source', 'extent_config'])
+    for (benchmark_source, extent_configuration), benchmark_metrics in benchmark_by_source:
+
+        # Define and create the output workspace as a subfolder within the supplied workspace
+        output_workspace = Path(workspace) / benchmark_source / extent_configuration.lower()
+        output_workspace.mkdir(parents = True, exist_ok = True)
+        output_png = Path(output_workspace) / f"{benchmark_source}_{extent_configuration.lower()}_stackedbar{'_indiv'if individual_plots else ''}.png"
+        if diff_stat:
+            output_png = Path(output_workspace) / f"{benchmark_source}_{extent_configuration.lower()}_diff_plot_{diff_stat}.png"
+            diff_bar_plots(versions, metric_csv, category=benchmark_source, outfig=output_png, stat=diff_stat)
+        elif individual_plots:
+            output_png = Path(output_workspace) / f"{benchmark_source}_{extent_configuration.lower()}_stackedbar_indiv.png"
+            eval_plot_stack_indiv(metric_csv=metric_csv, versions=versions, category=benchmark_source, outfig=output_png)
+        else:
+            output_png = Path(output_workspace) / f"{benchmark_source}_{extent_configuration.lower()}_stackedbar.png"
+            eval_plot_stack(metric_csv=metric_csv, versions=versions, category=benchmark_source, show_iqr=show_iqr, outfig=output_png)
+
+
+if __name__ == '__main__':
+    # Parse arguments
+    parser = argparse.ArgumentParser(description = f'Plot and aggregate statistics for benchmark datasets (BLE/AHPS libraries)')
+    parser.add_argument('-m','--metric_csv', help = 'Metrics csv created from synthesize test cases.', required = True)
+    parser.add_argument('-w', '--workspace', help = 'Output workspace', required = True)
+    parser.add_argument('-v', '--versions', help = 'List of versions to be plotted/aggregated. Versions are filtered using the "startswith" approach. For example, ["fim_","fb1"] would retain all versions that began with "fim_" (e.g. fim_1..., fim_2..., fim_3...) as well as any feature branch that began with "fb". An other example ["fim_3","fb"] would result in all fim_3 versions being plotted along with the fb.', nargs = '+', default = [])
+    parser.add_argument('-i', '--site-plots', help = 'If enabled individual barplots for each site are created.', action = 'store_true', required = False)
+    parser.add_argument('-iqr', '--show-iqr', help = 'If enabled, inter-quartile range error bars will be added.', action = 'store_true', required = False)
+    parser.add_argument('-d', '--diff-stat', help = 'This option creates diff plots instead of stacked bar plots. Only 2 versions can be used with this option. ' + \
+        "Input one of the following statistics to be used for comparison: ('CSI', 'MCC', 'TPR', 'PND', 'FAR').", default = False, required = True)
```

Comment:
@CarsonPruitt-NOAA I believe this argument should be "optional". Throws an error if you leave it out. I made the change and tested it successfully.

Reply:
Ah good catch... go ahead and push your fix if you've already coded it up.